### ISSUE TITLE
Support multiple backlight targets via multiple PWM outputs

### ISSUE THREAD

**Issue by HMarxen** [KMKfw/kmk_firmware#273]
One feature I always wished for in QMK was having more than one key backlight zones. Since KMK compatible chips are generally more modern, many also have more PWM pins. So, this feature would make even more sense in KMK. So far, anything mor complex than one backlight zone would need an LED driver IC. With more supported individual PWM outputs, you could get away with a lot more experimentation before you need to add special ICs.

You could for example adjust the brightness of 7-segment displays independent from key BL. Or using an analogue volt meter, the kind with a moving needle, as a typing speed gauge. Or just having the wasd keys and other gaming relevant keys adjusted separately.

I would suggest the following behaviour:
Instead of one pin you can define an array of PWM pins.
KC.LED_INC, KC.LED_DEC and KC.LED_TOG etc. control all PWM pins and sets them back to the same values if they deviate.
KC.LED_INC(x) etc. controls individual PWM pins. X being the array position of the PWM pin.

**Comment by HMarxen** [comment]
The RP2040 has 29 GPIO pins and 16 of those can do PWM. A numpad could have an individually adressable LED in each key. You could make a full light show without even needing Transistors.

**Comment by TobiasWegener** [comment]
> 

I did try it and the RB Pico RP2040 supports sufficient current to support all 16 LED of my NumPad in full brightness, without any additional hardware (apart from 470 Ohm resistors).
But I had to use the CircuitPython internal library  (digitalio) to drive the Leds because I could not get the KMK implementation running.
Maybe I am to stupid to understand the documentation [LED](https://github.com/KMKfw/kmk_firmware/blob/master/docs/led.md) but I find it pretty limited and can't get it to work.

Is there a simple and complete code example for mono colored Led, that works out of the box for a Pico with Circuitpython 7.0 and up-to-date kmk_firmware? Thanks a lot.
